app/src/main/java/de/westnordost/streetcomplete/data/CacheTrimmer.kt

```diff
@@ -0,0 +1,19 @@
+package de.westnordost.streetcomplete.data
+
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataController
+import de.westnordost.streetcomplete.data.quest.VisibleQuestsSource
+
+class CacheTrimmer(
+    private val visibleQuestsSource: VisibleQuestsSource,
+    private val mapDataController: MapDataController,
+) {
+    fun trimCaches() {
+        visibleQuestsSource.trimCache()
+        mapDataController.trimCache()
+    }
+
+    fun clearCaches() {
+        mapDataController.clearCache()
+        visibleQuestsSource.clearCache()
+    }
```

Comment:
Is there a reason why the calls are in a different order in these functions? If so, please add a comment. Otherwise, it's fine as is.

Reply:
No, there is no reason and the caches don't interact, so order should not matter.

Though for `trimCaches` the current order may be slightly preferrable over reverse order, as `mapDataController.trimCache()` is relatively slow and creates some intermediate lists, which may not be good when memory is low. So freeing some memory by `visibleQuestsController` first may be helpful here.

But since this happens on memory_low, and not memory_critical, I guess the difference will be negligible.

Reply:
I changed the order to be consistent.
The intermediate lists in `MapDataCache.trim` are actually created _after_ dropping large parts of the `spatialCache`, so we should already have enough memory available anyway.